AdversarialSearch/Minimax/TicTacToeMinimax.cpp

```diff
@@ -0,0 +1,203 @@
+};
+
+std::vector<int> TicTacToeMinimax::evaluateMaxDecisions(int board)
+{
+    std::vector<int> bestMoves;
+    int bestMoveScore = INT_MIN;
+    int mask = 3;
+    
+    for (int x = 0; x < 9; x++)
+    {
+        if ((board & mask) == 0)
+        {
+            int newBoard = (board | mask);
+            int score = getMinTreeScore(newBoard, -100, 100);
+            
+            if (score > bestMoveScore)
+            {
+                bestMoveScore = score;
+                
+                bestMoves.clear();
+                bestMoves.push_back(newBoard);
+            }
+            else if (score == bestMoveScore)
+            {
+                bestMoves.push_back(newBoard);
+            }
+        }
+        mask <<= 2;
+    }
+    return bestMoves;
+};
+
+std::vector<int> TicTacToeMinimax::evaluateMinDecisions(int board)
+{
+    std::vector<int> bestMoves;
+    int bestMoveScore = INT_MAX;
+    int mask = 2;
+    
+    for (int x = 0; x < 9; x++)
+    {
+        if ((board & mask) == 0)
+        {
+            int newBoard = (board | mask);
+            int score = getMaxTreeScore(newBoard, -100, 100);
+            
+            if (score < bestMoveScore)
+            {
+                bestMoveScore = score;
+                
+                bestMoves.clear();
+                bestMoves.push_back(newBoard);
+            }
+            else if (score == bestMoveScore)
+            {
+                bestMoves.push_back(newBoard);
+            }
+        }
+        mask <<= 2;
+    }
+    return bestMoves;
+};
+
+int TicTacToeMinimax::evaluateMaxDecision(int board)
+{
+    int bestMove = 0;
+    int bestMoveScore = INT_MIN;
+    int mask = 3;
+    
+    for (int x = 0; x < 9; x++)
+    {
+        if ((board & mask) == 0)
+        {
+            int score = getMinTreeScore(board | mask, -100, 100);
+
+            if (score > bestMoveScore)
+            {
+                bestMoveScore = score;
+                bestMove = (board | mask);
+            }
+        }
+        mask <<= 2;
+    }
+    return bestMove;
+};
+
+int TicTacToeMinimax::evaluateMinDecision(int board)
+{
+    int bestMove = 0;
+    int bestMoveScore = INT_MAX;
+    int mask = 2;
+    
+    for (int x = 0; x < 9; x++)
+    {
+        if ((board & mask) == 0)
+        {
+            int score = getMaxTreeScore(board | mask, -100, 100);
+
+            if (score < bestMoveScore)
+            {
+                bestMoveScore = score;
+                bestMove = (board | mask);
+            }
+        }
+        mask <<= 2;
+    }
+    return bestMove;
+};
+
+int TicTacToeMinimax::getMaxTreeScore(int board, int alpha, int beta)
+{
+    int state = getState(board);
+    if (state != Playing) return getScore(state);
+    
+    int bestScore = INT_MIN;
+    int mask = 3;
+
+    for (int x = 0; x < 9; x++)
+    {
+        if ((board & mask) == 0)
+        {
+            int score = getMinTreeScore(board | mask, alpha, beta);
```

Comment:
I suggest make this bitshift operation a constant variable inside the loop so it is easier to debug its value